packages/template-retail-react-app/app/components/store-locator-modal/store-locator-content.jsx

```diff
@@ -5,21 +5,133 @@
 import {
     Heading,
     Accordion,
     AccordionItem,
-    Box
+    Box,
+    Button
 } from '@salesforce/retail-react-app/app/components/shared/ui'
 import StoresList from '@salesforce/retail-react-app/app/components/store-locator-modal/stores-list'
 import StoreLocatorInput from '@salesforce/retail-react-app/app/components/store-locator-modal/store-locator-input'
-import {STORE_LOCATOR_DISTANCE} from '@salesforce/retail-react-app/app/constants'
 
-const StoreLocatorContent = ({form, submitForm, storesInfo, searchStoresParams}) => {
+// Others
+import {
+    SUPPORTED_STORE_LOCATOR_COUNTRIES,
+    DEFAULT_STORE_LOCATOR_COUNTRY,
+    STORE_LOCATOR_DISTANCE,
+    STORE_LOCATOR_NUM_STORES_PER_LOAD,
+    STORE_LOCATOR_DISTANCE_UNIT
+} from '@salesforce/retail-react-app/app/constants'
+const NUM_STORES_PER_REQUEST_API_MAX = 200
+
+// Hooks
+import {useSearchStores} from '@salesforce/commerce-sdk-react'
+import {useForm} from 'react-hook-form'
+const useGeolocation = (
+    setSearchStoresParams,
+    userHasSetManualGeolocation,
+    setUserHasSetManualGeolocation,
+    setAutomaticGeolocationHasFailed
+) => {
+    const getGeolocationError = () => {
+        setAutomaticGeolocationHasFailed(true)
+    }
+
+    const getGeolocationSuccess = (position) => {
+        setAutomaticGeolocationHasFailed(false)
+        setSearchStoresParams({
+            latitude: position.coords.latitude,
+            longitude: position.coords.longitude,
+            limit: STORE_LOCATOR_NUM_STORES_PER_LOAD
+        })
+    }
+
+    const getUserGeolocation = () => {
+        if (typeof navigator !== 'undefined' && navigator.geolocation) {
+            navigator.geolocation.getCurrentPosition(getGeolocationSuccess, getGeolocationError)
+            setUserHasSetManualGeolocation(false)
+        } else {
+            console.log('Geolocation not supported')
+        }
+    }
+
+    useEffect(() => {
+        if (!userHasSetManualGeolocation) getUserGeolocation()
+    }, [])
+
+    return getUserGeolocation
+}
+
+const StoreLocatorContent = ({
+    searchStoresParams,
+    setSearchStoresParams,
+    userHasSetManualGeolocation,
+    setUserHasSetManualGeolocation
+}) => {
     const intl = useIntl()
+    const form = useForm({
+        mode: 'onChange',
+        reValidateMode: 'onChange',
+        defaultValues: {
+            countryCode: userHasSetManualGeolocation ? searchStoresParams.countryCode : '',
+            postalCode: userHasSetManualGeolocation ? searchStoresParams.postalCode : ''
+        }
+    })
+    const [automaticGeolocationHasFailed, setAutomaticGeolocationHasFailed] = useState(false)
+    const [userWantsToShareLocation, setUserWantsToShareLocation] = useState(false)
+
+    const getUserGeolocation = useGeolocation(
+        setSearchStoresParams,
+        userHasSetManualGeolocation,
+        setUserHasSetManualGeolocation,
+        setAutomaticGeolocationHasFailed
+    )
+    const [limit, setLimit] = useState(searchStoresParams.limit)
```

Comment:
How about using another name other than `limit`? Maybe `numStoresToShow`?

I found myself confused with the limit we passed to the API vs this soft limit for our UI. I think it's better if we can be distinguish those 2 things.